modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuilder.java

```diff
@@ -133,71 +176,77 @@ private BuildIndexTask(
 
         @Override
         public void run() {
```

Comment:
This future is really hard to read, maybe it can be somehow split into separate methods? I am afraid of it turning out like the TableManager

Reply:
It seems that there is not so much code, I tried to add one more method.